product_cost_usd/models/product_template.py

```diff
@@ -0,0 +1,42 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import float_compare
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    standard_price_usd = fields.Float(
+        "Cost in USD", digits="Product Price", help="Price cost of the product in USD currency"
+    )
+
+    @api.constrains("standard_price_usd", "seller_ids")
+    def check_cost_and_price(self):
+        """Validate 'Cost in USD' usability.
+
+        Usability conditions:
+        - Before set a 'Cost in USD' in a product at least one supplier should
+          have price in USD.
+        - The Cost in USD cannot be less than supplier price.
+        """
+        usd_currency = self.env.ref("base.USD")
+        prec = self.env["decimal.precision"].precision_get("Product Price")
+        for product in self:
+            usd_seller = product.seller_ids.filtered(lambda x: x.currency_id == usd_currency)
```

Comment:
```suggestion
            usd_seller = product.seller_ids.filtered(lambda x: x.currency_id == usd_currency)[:1]
```

But maybe `_select_seller` may be used.